src/crimson/os/seastore/seastore.cc

```diff
@@ -347,8 +347,9 @@ SeaStore::omap_get_values(
   const ghobject_t &oid,
   const omap_keys_t &keys)
 {
-  auto c = static_cast<SeastoreCollection*>(ch.get());
-  return repeat_with_onode<omap_values_t>(
+  return seastar::do_with(oid, keys, [this, ch = std::move(ch)] (auto &oid, auto &keys) {
```

Comment:
the commit message does not explain why we need this change.

and the `repeat_with_onode()` also use seastar::do_with()` to capture `oid`. why do you think it's different from the `do_with()` call here?

i think we should instead capture `keys` in the continuation passed to `repeat_with_onode()`. that continuation is kept alive before `repeat_with_onode()` completes.

Reply:
done. 

" the commit message does not explain why we need this change."  what do you suggest?

Reply:
> done.
> 
> " the commit message does not explain why we need this change." what do you suggest?

the latest revision looks good.